instrumentation/liberty/liberty/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/liberty/LibertyStartSpanAdvice.java

```diff
@@ -26,5 +26,9 @@ public static void onEnter() {
 
     ctx.setContext(context);
     ctx.setScope(scope);
+
+    // For startAsync instrumentation to work on Liberty, the listener needs to be added with a
+    // response object included, which is not available in that instrumentation.
```

Comment:
it seems like we should always be passing request/response to AsyncContext.addListener(), since our listener calls [AsyncEvent.getSuppliedRequest()](https://docs.oracle.com/javaee/7/api/javax/servlet/ServletRequest.html#startAsync--) and [AsyncEvent.getSuppliedResponse()](https://docs.oracle.com/javaee/7/api/javax/servlet/AsyncEvent.html#getSuppliedResponse--).

-- or --

maybe our listener should call `event.getAsyncContext().getRequest()` and  `event.getAsyncContext().getResponse()` instead, and then maybe we don't ever need to pass request/response?

Reply:
`event.getAsyncContext().getResponse()` does not seem to solve the current issue with Liberty as it is not available there either (and the spec allows it to be missing). Passing the response would mean storing the response as a request attribute for all servlet engines - I am not sure if we want to do that preemptively for all of them.

Reply:
If we don't pass request/response to AsyncContext.addListener(), won't AsyncEvent.getSuppliedResponse() (which we call in our listener) always return null?

> If the AsyncListener was added via AsyncContext.addListener(AsyncListener), this method must return null.

https://docs.oracle.com/javaee/7/api/javax/servlet/AsyncEvent.html#getSuppliedResponse--